pkg/reconciler/autoscaling/kpa/kpa.go

```diff
@@ -114,10 +114,16 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pa *autoscalingv1alpha1.
 	if err := c.ReconcileMetric(ctx, pa, resolveScrapeTarget(ctx, pa)); err != nil {
 		return fmt.Errorf("error reconciling Metric: %w", err)
 	}
+	podCounter := resourceutil.NewPodAccessor(c.podsLister, pa.Namespace, pa.Labels[serving.RevisionLabelKey])
+
+	pod, err := podCounter.GetAnyPod()
+	if err != nil {
+		return fmt.Errorf("error getting a pod for the revision: %w", err)
+	}
```

Comment:
Fetching a pod here seems premature

Reply:
Could you elaborate? We already do that here via the pod accessor.